File: Hibernate & JPA/Code/03/Z tabelą łącznikową/Supplier.java
package org.example;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
public class Supplier {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int supplierID;
    private String companyName;
    private String street;
    private String city;

    @OneToMany
    private final List<Product> products = new ArrayList<>();

    public Supplier() {}

    public Supplier(String companyName, String street, String city) {
        this.companyName = companyName;
        this.street = street;
        this.city = city;
    }

    public void addProducts(List<Product> products) {
        this.products.addAll(products);
    }
}

File: Entity Framework/Kod/03/Supplier.cs
namespace ŁukaszDydekEFProducts
{
    internal class Supplier
    {
        public int SupplierID { get; set; }
        public string CompanyName { get; set; }
        public string Street { get; set; }
        public string City { get; set; }
        public List<Product> Products { get; set; } = new();

        public Supplier(string CompanyName, string Street, string City) 
        { 
            this.CompanyName = CompanyName;
            this.Street = Street;
            this.City = City;
        }

        public void AddProducts(List<Product> products)
        {
            Products.AddRange(products);
        }
    }
}
